### PR TITLE
feat (react component) : Add SetFlexibleControlsType to the top bar

### PR DESCRIPTION
**Pull request by nilscognite**
image

#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:
Make buttons for changing controls type.

NOTE: I have added possibility to the third control type also, This is not in use, but I keep it for completeness.
## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/RevealToolbar/SetFlexibleControlsTypeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      return 'Circle';
+    case FlexibleControlsType.OrbitInCenter:
+      return 'Coordinates';
+    default:
+      return 'Error';
+  }
+}
+
+function asFlexibleCameraManager(manager: CameraManager): IFlexibleCameraManager | undefined {
+  // instanceof don't work within React, so using safeguarding
```
Comment (haakonflatval-cognite) [inline react-components/src/components/RevealToolbar/SetFlexibleControlsTypeButton.tsx:120]: I think the reason is that `instanceof` doesn't work with checking if it's an instance of an interface, only if it's an instance of a class

Comment (nilscognite) [reply react-components/src/components/RevealToolbar/SetFlexibleControlsTypeButton.tsx:121]: Promod mentioned that React could handle it.

**Comment by haakonflatval-cognite** [comment]
Title should be `feat(react-components): ....`

File: viewer/packages/camera-manager/src/Flexible/FlexibleControlsTranslator.ts [hunk 2]
```diff
@@ -51,8 +51,8 @@ export class FlexibleControlsTranslator {
     if (!this._camera) {
       return false;
     }
-    const normalizedCoords = this._controls.getNormalizedPixelCoordinates(event);
```
[the remark this answers is not in the document]
Comment (nilscognite) [reply viewer/packages/camera-manager/src/Flexible/FlexibleControlsTranslator.ts:54]: This was simply a mistake from my side. It was a file that wasn't saved in another branch actually.

**Review by haakonflatval-cognite — APPROVED** [review]

File: react-components/src/components/RevealToolbar/SetFlexibleControlsType.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  includeOrbitInCenter
+}: CustomSettingsProps): ReactElement {
+  const viewer = useReveal();
+  const manager = asFlexibleCameraManager(viewer.cameraManager);
+  const { t: translate } = useTranslation();
+
+  // This value is redunant (react force me to do it)
+  const [selectedValue, setSelectedValue] = useState<FlexibleControlsType>(
+    getDefaultValue(manager)
+  );
```
Comment (haakonflatval-cognite) [inline react-components/src/components/RevealToolbar/SetFlexibleControlsType.tsx:35]: Could we have somewhat more descriptive variable names?

Comment (nilscognite) [reply react-components/src/components/RevealToolbar/SetFlexibleControlsType.tsx:35]: mmm. It's only one thing that can be selected here

Comment (nilscognite) [reply react-components/src/components/RevealToolbar/SetFlexibleControlsType.tsx]: This is now fixed

**Pull request merged**